src/github.com/matrix-org/dendrite/federationapi/routing/profile.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2017 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package routing
+
+import (
+	"net/http"
+
+	"github.com/matrix-org/dendrite/clientapi/auth/storage/accounts"
+	"github.com/matrix-org/dendrite/clientapi/httputil"
+	"github.com/matrix-org/dendrite/clientapi/jsonerror"
+	clientRouting "github.com/matrix-org/dendrite/clientapi/routing"
```

Comment:
It feels weird that we're importing the routing for another component here. I think the best bet would be to move the structs to `src/github.com/matrix-org/dendrite/common/types.go` and then you can import common.

Reply:
This was fixed as well.